### ISSUE TITLE
Internal Server Error 9.0.2 (I think its parsing of my contacts)

### ISSUE THREAD

**Issue by ralin3** [skjnldsv/Owncloud-FBSync#25]
Hi,
When pressing "Match contacts" and/or "Sync matched contacts" I get the "Internal Server Error" from Owncloud, server.log (http://pastebin.com/T1k3Jd22)

This didnt work with 9.0.1 from packages, and still doesn't work after 9.0.2
Reading the log, Im sure it is because of a corrupt contact or something.
(Davdroid is also throwing me errors on my phone regarding a object.)

This is obviously something wrong on my end, but the errormessage could maybe be a bit better/obvious?

**Comment by skjnldsv** [comment]
Hello! :)
You guessed right, this is because o one of your contact that doesn't follow the vcard requirements, aka corrupted! :(
The problem is the log being truncated. On the line `Parser\MimeDir->parse('BEGIN:VCARD\r...` if you had the full log, you will be able to get the contact data :)

**Comment by ralin3** [comment]
How could I get the full log? I thought I had that already.

**Comment by ralin3** [comment]
I managed to fix my issue by restoring a backup, but could it be possible to create a better errormessage of sort?
Or is it limited to what owncloud(-core) can give you?

**Comment by skjnldsv** [comment]
This is a good question!
I will probably need to do something yes! :)

**Comment by ralin3** [comment]
So, I managed to "track" this down.
It is actually because of FBSync - I can reproduce it 100% of the time when trying to write to a few contacts which don't have any pictures at all before matching them.

All my other contacts had pictures before, except two - when I match those two and sync, everything related to contacts crash.

Is there any way I can provide some logs or something so we can track this down?

**Comment by ralin3** [comment]
Anyway I can assist more with tracking this issue down?
I tried adding a random picture to a contact, and then overwriting the picture from facebook, but the same issue happens.

**Comment by skjnldsv** [comment]
Well, to be fair, I don't have the time to manage and fix this app! It works on most of the people who tried it, but some (like you) have some issues.
I'm sorry! :/

**Comment by ralin3** [comment]
I was a bit annoyed with this, so I started to dig deeper.

I did not find the root cause, but at least I found a workaround.

Apparently there is something wrong with the contacts I create on my phone and then sync.
I noticed that contacts created in OC had alot more fields available for editing than the ones created on my phone.
I therefore recreated the contacts in OC, and deleted the ones I had made on the phone - now sync started to work again.

If anyone else have the same issue, when you are in the "Match Contacts"-window in FBsync, if the profile picture is "blank"/white with a spinning load-circle - that contact needs to be recreated.
